embrace-test-fakes/src/main/kotlin/io/embrace/android/embracesdk/fakes/FakeMainThreadHandler.kt

```diff
@@ -0,0 +1,14 @@
+package io.embrace.android.embracesdk.fakes
+
+import io.embrace.android.embracesdk.internal.handler.MainThreadHandler
+
+class FakeMainThreadHandler : MainThreadHandler {
+    override fun postAtFrontOfQueue(function: () -> Unit) {
+        function()
+    }
+
+    override fun postDelayed(runnable: Runnable, delayMillis: Long) {
+        Thread.sleep(1000L)
+        runnable.run()
```

Comment:
IMO you can probably remove the sleep unless it's necessary for testing, and even then, I'd prefer using a CountdownLatch or very small delay (e.g. 1ms) as anything else will unnecessarily slow down the entire test suite

Reply:
yeah, I wasn't too sure about this, but I wanted postDelayed to run after postAtFrontQueue